### PR TITLE
Fix image aspect settings MacOS

### LINKED ISSUE #7949: [Bug] Aspect settings not working on MacOS (4.3-pre)
### Description
Only 1 of the 3 aspect settings works on XF Mac.

### Steps to Reproduce

1. Create a Xamarin Forms project from template
2. Add the Mac project to it as per document: https://docs.microsoft.com/en-us/xamarin/xamarin-forms/platform/other/mac
3. Add image controls to the main page with each aspect setting being used.
4. Run the app. 

### Expected Behavior
ImageAspectiOS

### Actual Behavior
ImageAspectMac

### Basic Information

- Version with issue: 4.3-pre3
- Last known good version: ?, but this appears to be a recent major refactor that would be my guess at when it broke: https://github.com/xamarin/Xamarin.Forms/commit/ce148bf4c81f7d1da5ef74c7a2a43cca6ca73331#diff-59032a35e10fb51bc88a456719271ac1R1-R12. I think it may have lost some things (see my comment at the end of that link).
- IDE: VS on Mac

### Reproduction Link
[ImageAspect.zip](https://github.com/xamarin/Xamarin.Forms/files/3717725/ImageAspect.zip)

### PR DESCRIPTION
**Pull request by jfversluis**
### Description of Change ###

Fix the different aspect settings for `Image` on MacOS.

### Issues Resolved ### 

- fix #7949

### API Changes ###
 
 None

### Platforms Affected ### 

- MacOS

### Behavioral/Visual Changes ###

Make the different image aspect settings work again for MacOS. Seems working is broken after a big refactoring.

### Before/After Screenshots ### 

Before

image

After

image

### Testing Procedure ###

Added a issue7949 page to the gallery app. If all images show a different aspect setting, it worked!

I've added a box in the lower right with a "cycle" button that lets you change the aspect to see if they all work and if they work when changed at runtime.

### PR Checklist ###

- [x] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7949.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+			await Task.Delay(1000);
+			RunningApp.Screenshot("Lower right should show AspectFill");
+
+			RunningApp.Tap(CycleButtonAutomationId);
+			await Task.Delay(1000);
+			RunningApp.Screenshot("Lower right should show Fill");
+
+			RunningApp.Tap(CycleButtonAutomationId);
+			await Task.Delay(1000);
+			RunningApp.Screenshot("Lower right should show AspectFit");
```
Comment (felipebaltazar) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7949.cs]: You can validate some properties on MacOS to previne regression.

Something like that:

```C#

#if __MACOS__
var nsImageScale = RunningApp.Query(c => c.Marked(YOUR_IMAGE_AUTOMATIONID).Invoke("ImageScaling ").Value<int>());

Assert.AreEqual(1, nsImageScale );
#endIf
```

[Documentation](https://docs.microsoft.com/en-us/dotnet/api/xamarin.uitest.queries.appquery.invoke?view=xamarin-uitest-sdk)

[My pr with similar test](https://github.com/xamarin/Xamarin.Forms/pull/7966/files#diff-c8584e3f1ae3f5ab5220d09f4873ea1aR69)

[StackOverflow question](https://stackoverflow.com/questions/45878350/xamarin-ui-test-take-controls-background-color-ios)

[Xamarin Forum question](https://forums.xamarin.com/discussion/61782/check-button-color-with-xamarin-uitest)

File: Xamarin.Forms.Platform.iOS/Renderers/ImageElementManager.cs [hunk 2]
```diff
@@ -80,6 +80,27 @@ public static void SetAspect(IImageVisualElementRenderer renderer, IImageElement
...
 			Control.Layer.ContentsGravity = imageElement.Aspect.ToNSViewContentMode();
+
+			switch (imageElement.Aspect)
+			{
+
+				case Aspect.AspectFill:
+					// No built in aspect fill on the NSImageView directly, it is there on the layer so go through there
+					if (Control.Image != null)
+					{
+						Control.Layer = new CALayer();
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.iOS/Renderers/ImageElementManager.cs]: We should check if we have a Layer before creating one, the ViesualElementTracker set s layer properties.

if( Control.Layer != null)
    Control.Layer = new CALayer();

Comment (jfversluis) [reply Xamarin.Forms.Platform.iOS/Renderers/ImageElementManager.cs]: Do you mean `if( Control.Layer != null)` like you said or `if( Control.Layer == null)` to just create a Layer once?

Comment (rmarinho) [reply Xamarin.Forms.Platform.iOS/Renderers/ImageElementManager.cs]: right if null :P 

getting back from vacations :)

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Xamarin.Forms.Controls.Issues.Shared.projitems [hunk 3]
```diff
@@ -1500,7 +1501,7 @@
   <ItemGroup>
     <EmbeddedResource Include="$(MSBuildThisFileDirectory)Issue7593.xaml">
       <SubType>Designer</SubType>
-      <Generator>MSBuild:Compile</Generator>
+      <Generator>MSBuild:UpdateDesignTimeXaml</Generator>
```
Comment (kingces95) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Xamarin.Forms.Controls.Issues.Shared.projitems]: Is this intentional?

File: Xamarin.Forms.Platform.iOS/Renderers/ImageElementManager.cs (hunk 2 above)
Comment (jfversluis) [reply Xamarin.Forms.Platform.iOS/Renderers/ImageElementManager.cs]: So, the weird thing is when I implement that it stops working... 🤔

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Xamarin.Forms.Controls.Issues.Shared.projitems (hunk 3 above)
Comment (jfversluis) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Xamarin.Forms.Controls.Issues.Shared.projitems]: Nope! :( it happens automatically whenever I add something I think. Probably some difference between how VSWin and VSMac handles things. I don't think it has any impact but I can take it out if you want

File: Xamarin.Forms.Platform.iOS/Renderers/ImageElementManager.cs (hunk 2 above)
Comment (rmarinho) [reply Xamarin.Forms.Platform.iOS/Renderers/ImageElementManager.cs]: Ok so something wrong with this fix :P you need to investigate the work it's done on the tracker and this i think.

Comment (jfversluis) [reply Xamarin.Forms.Platform.iOS/Renderers/ImageElementManager.cs]: Yep, I'll mark it DNM for now and revisit later.

Comment (DennisWelu) [reply Xamarin.Forms.Platform.iOS/Renderers/ImageElementManager.cs]: Could be something related to when `wantsLayer = true` happens relative to a default backing layer getting created prior to this point? It is apparently [sensitive to order](https://stackoverflow.com/a/38752428/1735721).

**Comment by Suplanus** [comment]
Is there a plan when this PR is merged into 4.5? Would like to test it in the nightly.

**Comment by jfversluis** [comment]
Not right now :( there is still a problem with one of the aspects not playing nicely

**Comment by Suplanus** [comment]
Thanks for your reply!
Look at this every day because I cannot ship my mac version without this fix.

**Comment by jfversluis** [comment]
Sorry to hear that @Suplanus, sorry about that!

You wouldn't happen to have any idea on how to fix the last bit? That would probably speed things up :)

**Comment by samhouts** [comment]
I'm going to close this one for now until we're ready to return to it.

**Comment by Suplanus** [comment]
Why the issue can't stay open?

**Comment by DennisWelu** [comment]
Would be nice to have this one finished off, I say as a cheerleader. It's 3/4 "done" to fix the Aspect setting on Mac.

**Comment by Suplanus** [comment]
I also think that image aspects are fundamental for Xamarin.Forms.MacOs...

**Pull request closed without merging**